clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy

```diff
@@ -425,7 +415,13 @@ class BasicGoogleDeployHandler implements DeployHandler<BasicGoogleDeployDescrip
         }
         namedPorts = sourceServerGroup?.namedPorts?.collect { name, port -> new NamedPort(name: name, port: port) }
       } else {
-        namedPorts = description?.loadBalancingPolicy?.namedPorts
+        def loadBalancingPolicy = description?.loadBalancingPolicy
+        if (loadBalancingPolicy?.namedPorts)  {
```

Comment:
Right here I'm worried you could get into a weird state if `namedPorts` is an empty list and `listeningPort` is still defined on the loadBalancingPolicy.

I would do an explicit `if (loadBalancingPolicy?.namedPorts != null) { ... }`.